core/src/main/java/com/linecorp/armeria/client/DefaultWebClient.java

```diff
@@ -44,36 +39,33 @@
 
     @Override
     public HttpResponse execute(HttpRequest req) {
-        URI uri;
-
-        if (isAbsoluteUri(req.path())) {
-            try {
-                uri = URI.create(req.path());
-            } catch (Exception ex) {
-                logger.warn("Failed to create URI: {}", req.path(), ex);
-                uri = null;
-            }
-        } else {
-            if (req.scheme() != null && req.authority() != null) {
+        if (Clients.isUndefinedUri(uri())) {
+            final URI uri;
+            if (isAbsoluteUri(req.path())) {
+                try {
+                    uri = URI.create(req.path());
+                } catch (Exception ex) {
+                    return abortRequestAndReturnFailureResponse(req, new IllegalArgumentException(
+                            "Failed to create a URI: " + req.path(), ex));
+                }
+            } else if (req.scheme() != null && req.authority() != null) {
                 uri = req.uri();
             } else {
-                uri = null;
+                return abortRequestAndReturnFailureResponse(req, new IllegalArgumentException(
+                        "A URI with scheme and authority must be specified. path: " + req.path()));
```

Comment:
Realized that a user can specify scheme and authority in headers. How about this:

```
"Scheme and authority must be specified in \":path\" or in \":scheme\" and \":authority\". :path=" + 
req.path() + ", :scheme=" + req.scheme() + ", :authority=" + req.authority()
```

Reply:
Fixed. 😉